### PR TITLE
Add a summary for disaggregated analysis showing ratio and difference

### PR DESCRIPTION
**Pull request by romanlutz**
## Description

This PR adds another table with difference and ratio for all metrics. This includes a tooltip. Unlike the previous two tables this is not a heatmap with colored backgrounds. Instead, all fields are transparent.

In addition to the summary table, this PR also
- puts all y-axis labels in a grid (as an extension of the table) and aligns all those label columns so that there are no misaligned columns for the same metrics anymore.
- adds N/A as a shown value for undefined metrics (when, for example, FPR can't be computed due to lack of negatives in the data)
- fixes a bug in the disaggregated analysis code where the wrong cohorts were used to calculate the metrics

Tests will be added before the feature flag is removed (separate PR)

https://user-images.githubusercontent.com/10245648/164485416-83f10e83-dd0f-4892-a565-7fed66536ad9.mp4

## Areas changed

npm packages changed:

- [ ] responsibleai/causality
- [ ] responsibleai/core-ui
- [ ] responsibleai/counterfactuals
- [ ] responsibleai/dataset-explorer
- [ ] responsibleai/fairness
- [ ] responsibleai/interpret
- [x] responsibleai/localization
- [ ] responsibleai/mlchartlib
- [x] responsibleai/model-assessment

Python packages changed:

- [ ] erroranalysis
- [ ] raiutils
- [ ] raiwidgets
- [ ] rai_core_flask
- [ ] responsibleai

## Tests

- [ ] No new tests required.
- [ ] New tests for the added feature are part of this PR.
- [ ] I validated the changes manually.

## Documentation:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by tongyu-microsoft — APPROVED** [review]

**Comment by tongyu-microsoft** [comment]
From UI/user point of view, this tooltip might be too long to read. Could we limit the width?
image

**Comment by romanlutz** [comment]
> From UI/user point of view, this tooltip might be too long to read. Could we limit the width? image

Excellent point! Let me see if they have some configuration option for that. Otherwise I'll try to hack it.

File: .eslintrc/.eslintrc.comments.eslintrc [hunk 1]
```diff
@@ -2,6 +2,9 @@
   "plugins": ["eslint-plugin-eslint-comments"],
   "extends": ["plugin:eslint-comments/recommended"],
   "rules": {
-    "eslint-comments/no-use": ["error", { "allow": [] }]
+    "eslint-comments/no-use": [
+      "error",
+      { "allow": ["eslint-disable-next-line"] }
```
Comment (xuke444) [inline .eslintrc/.eslintrc.comments.eslintrc:7]: we intend to disable this for people not abuse this inline disable.

Do you mind create a variable for null in high chart folder and disable rule in eslintrc under that package folder?

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx [hunk 2]
```diff
@@ -115,7 +96,7 @@ export class CohortStatsHeatmap extends React.Component<
                 // make metric name lower case in sentence
                 this.series.xAxis.categories[this.point.x].toLowerCase(),
                 this.series.yAxis.categories[this.point.y],
-                pointValue
+                pointValue === null ? "N/A" : pointValue
```
Comment (xuke444) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx:99]: Is this visiable to user? Probably need localize

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx [hunk 3]
```diff
@@ -132,11 +113,11 @@ export class CohortStatsHeatmap extends React.Component<
               columns: [
                 {
                   labels: {
-                    // format labels to cap the line length
                     formatter() {
-                      return wrapYAxisLabels(this.value.toString(), true);
+                      const text = wrapYAxisLabels(this.value.toString());
+                      return `<div style='width:300px'>${text}</div>`;
```
Comment (xuke444) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx:118]: Does this formatter only take html string? not react element?

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FairnessMetricTable.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+                .fairnessMetricDifference,
+              localization.ModelAssessment.ModelOverview.fairnessMetricRatio
+            ],
+            grid: {
+              borderWidth: 2,
+              columns: [
+                {
+                  labels: {
+                    formatter() {
+                      return `<div style='width:300px'><p>${this.value}</p></div>`;
```
Comment (imatiach-msft) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FairnessMetricTable.tsx:156]: nit: should style be moved into styles file here?
```
style='width:300px'
```

**Review by imatiach-msft — APPROVED** [review]

**Comment by romanlutz** [comment]
@tongyu-microsoft new tooltip with wrapped text! I decided to make the cohort names bold to help with readability.

Tooltip for count column (special case because the values are integers rather than floating point numbers):
image

Tooltip for metric columns:
image

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FairnessMetricTable.tsx (hunk 4 above)
Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/FairnessMetricTable.tsx:156]: In general, I agree 100%.

In this particular case, you may have noticed that it's actually a string, not your usual React JSX. I briefly tried doing this with a style file, but it didn't work. I suspect that it has to do with this just being a string.

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx (hunk 3 above)
Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx:118]: Correct. You have two choices:
- a function (which is what I used) that returns a string
- just a format string (so not even a function) in a slightly differently named parameter

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx (hunk 2 above)
Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx:99]: Done

File: .eslintrc/.eslintrc.comments.eslintrc (hunk 1 above)
Comment (romanlutz) [reply .eslintrc/.eslintrc.comments.eslintrc:7]: Good idea! Will do.

**Review by vinuthakaranth — APPROVED** [review]

**Pull request merged**